src/cobra/core/reaction.py

```diff
@@ -177,29 +204,37 @@ def objective_coefficient(self):
         return linear_reaction_coefficients(self.model, [self]).get(self, 0)
 
     @objective_coefficient.setter
-    def objective_coefficient(self, value):
+    def objective_coefficient(self, value: float) -> None:
         if self.model is None:
             raise AttributeError("cannot assign objective to a missing model")
         if self.flux_expression is not None:
             set_objective(self.model, {self: value}, additive=True)
 
-    def __copy__(self):
+    def __copy__(self) -> "Reaction":
+        """Copy the Reaction."""
         cop = copy(super(Reaction, self))
         return cop
 
-    def __deepcopy__(self, memo):
+    # Unclear what memo should be
+    def __deepcopy__(self, memo) -> "Reaction":
+        """Copy the reaction with memo."""
         cop = deepcopy(super(Reaction, self), memo)
         return cop
 
     @staticmethod
-    def _check_bounds(lb, ub):
+    def _check_bounds(lb: Union[int, float], ub: Union[int, float]) -> None:
```

Comment:
As the input types for `lower_bound` and `upper_bound` are taken as `float`, we can use `float` for the argument types as well. Also, please add a Raises section in the docstring.